[Fix 8635] About showed "Version ()" instead of Version 0.14.0 (2019…)

MajorTomSec opened the pull request:
fixes #8635 

### Summary

This fixes #8635 by adding VERSION and BUILD_NUMBER files in the correct nix template and updating bash script in order to prevent it from failing, due to the git repository being not initialized in the nix environment.

#### Platforms
<!-- (Optional. Specify which platforms should be tested) -->

- Android
- iOS
- macOS
- Linux
- Windows

### Steps to test
- Open Status
- Create an account
- Open Profile -> About and check value shown for Version

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6e0827d6 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8821/1/) | 2019-08-22 17:26:48 | ~11 min | `ios` | [:package: ipa](https://i.diawi.com/LBokLL) |

jakubgs commented on scripts/build_no.sh:
```diff
@@ -23,12 +23,10 @@
-GIT_ROOT=$(git rev-parse --show-toplevel)
 DEV_BUILD_NUMBER=9999
-BUILD_NUMBER_FILE="${GIT_ROOT}/BUILD_NUMBER"
+BUILD_NUMBER_FILE="BUILD_NUMBER"
```
Wouldn't it make more sense to do something like:
```bash
if [[ -z "${BUILD_NUMBER_FILE}" ]]; then
  BUILD_NUMBER_FILE="${GIT_ROOT}/BUILD_NUMBER"
fi
```
And then provide the path of `BUILD_NUMBER_FILE` in Nix?
What do you think @PombeirP?

pedropombeiro replied:
I don't understand in which situation this script gets called and .git isn't there. Normally it is passed to Nix [here](https://github.com/status-im/status-react/blob/develop/ci/android.groovy#L38) where we have a Git repo. Unless it is happening [here](https://github.com/status-im/status-react/blob/develop/src/status_im/utils/build.clj#L25), but if that is the case, then I don't understand the need to add the `VERSION` to the jsbundle Nix expression, since it'll cause the bundle to be invalidated any time the `VERSION` file changes.

But yeah, I agree that passing an env var to the file in this situation is better as it still allows us having the script live anywhere and still have it behave correctly.

pedropombeiro commented on nix/targets/jsbundle.nix:
```diff
@@ -20,12 +20,12 @@ in stdenv.mkDerivation {
+          dirRootsToInclude = [
             "components/src" "react-native/src/cljsjs" "react-native/src/mobile" "src" "env/prod" "prod" # Taken from project.clj :profiles :prod :cljsbuild :builds :android :source-paths
-            "resources" "status-modules/cljs" "status-modules/resources"
+            "resources" "status-modules/cljs" "status-modules/resources" "scripts"
```
I'd maybe move the `scripts/gen_build_no.sh` and `scripts/build_no.sh` files to a subfolder (`scripts/version` or something), so that changes in unrelated scripts don't cause this Nix expression to be rebuilt.

pedropombeiro reviewed (changes requested):
Great job so far @MajorTomSec! Just added a couple questions/remarks.

MajorTomSec commented:
Thanks for the reviews.
I took into account all of your remarks and came up with this.

@PombeirP the issue is indeed with https://github.com/status-im/status-react/blob/develop/src/status_im/utils/build.clj#L25 and this is why the VERSION file has to be added to nix as well (see [here](https://github.com/status-im/status-react/blob/develop/src/status_im/utils/build.clj#L36)

I thought instead of using an env variable for specifying BUILD_NUMBER_FILE, I used pwd when git is not available:
`GIT_ROOT=$(git rev-parse --show-toplevel || pwd)`

I believe this way nothing would be broken. As build.clj uses relative path for VERSION and the script path, this would not fail here anyway.

Let me know if you have other remarks.

MajorTomSec committed:
[Fix 8635] About showed "Version ()" instead of Version 0.14.0 (201908...) on Android

Fixes #8635 by adding VERSION and BUILD_NUMBER files in the correct nix
template and updating bash script in order to prevent it from failing,
due to the git repository being not initialized in the nix environment.

Move scripts/build_no.sh and scripts/gen_build_no.sh to
scripts/version/build_no.sh to prevent Nix from rebuilding when
unrelated scripts are touched.

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

MajorTomSec commented:
I believe everything should be fine by now. 
Waiting for any other remark anyone might have.

jakubgs commented on scripts/version/build_no.sh:
```diff
@@ -26,12 +26,14 @@
 DEV_BUILD_NUMBER=9999
-BUILD_NUMBER_FILE="${GIT_ROOT}/BUILD_NUMBER"
+
+SCRIPTPATH="$(cd "$(dirname "$0")" ; pwd -P)"
```
That's okay, but I kinda like your suggestion of:
```
GIT_ROOT=$(git rev-parse --show-toplevel || pwd)
```
It's a nice trick, but this works too.

yenda commented:
@PombeirP seems like your request for changes has been fulfilled?

churik commented:
Checked on Android 8.0 and IOS 11.4.1
